src/integrations/prefect-redis/prefect_redis/blocks.py

```diff
@@ -0,0 +1,177 @@
+            raise ValueError("Missing hostname")
+        if self.username and not self.password:
+            raise ValueError("Missing password")
+
+    async def read_path(self, path: str) -> bytes:
+        """Read a redis key
+
+        Args:
+            path: Redis key to read from
+
+        Returns:
+            Contents at key as bytes
+        """
+        client = self.get_client()
+        ret = await client.get(path)
+
+        await client.close()
+        return ret
+
+    async def write_path(self, path: str, content: bytes) -> None:
+        """Write to a redis key
+
+        Args:
+            path: Redis key to write to
+            content: Binary object to write
+        """
+        client = self.get_client()
+        ret = await client.set(path, content)
+
+        await client.close()
+        return ret
+
+    def get_client(self) -> redis.Redis:
+        """Get Redis Client
+
+        Returns:
+            An initialized Redis async client
+        """
+        return redis.Redis(
+            host=self.host,
+            port=self.port,
+            username=self.username.get_secret_value() if self.username else None,
+            password=self.password.get_secret_value() if self.password else None,
+            db=self.db,
+            ssl=self.ssl,
+        )
+
+    def get_async_client(self) -> redis.asyncio.Redis:
+        """Get Redis Client
+
+        Returns:
+            An initialized Redis async client
+        """
+        return redis.asyncio.Redis(
+            host=self.host,
+            port=self.port,
+            username=self.username.get_secret_value() if self.username else None,
+            password=self.password.get_secret_value() if self.password else None,
+            db=self.db,
+            ssl=self.ssl,
+        )
+
+    @classmethod
+    def from_connection_string(
+        cls, connection_string: Union[str, SecretStr]
+    ) -> "RedisDatabase":
+        """Create block from a Redis connection string
+
+        Supports the following URL schemes:
+        - `redis://` creates a TCP socket connection
+        - `rediss://` creates a SSL wrapped TCP socket connection
+
+        Args:
+            connection_string: Redis connection string
+
+        Returns:
+            `RedisCredentials` instance
+        """
+        connection_kwargs = parse_url(
+            connection_string
+            if isinstance(connection_string, str)
+            else connection_string.get_secret_value()
+        )
+        ssl = connection_kwargs.get("connection_class") == redis.asyncio.SSLConnection
+        return cls(
+            host=connection_kwargs.get("host", "localhost"),
+            port=connection_kwargs.get("port", DEFAULT_PORT),
+            db=connection_kwargs.get("db", 0),
+            username=connection_kwargs.get("username"),
+            password=connection_kwargs.get("password"),
+            ssl=ssl,
+        )
+
+    def as_connection_params(self) -> Dict[str, Any]:
```

Comment:
This method makes it easier to configure a `RedisLockManager` from a block. An example of it's use in included in the `RedisLockManager` docstring.